packages/server/LeaderElector.ts

```diff
@@ -0,0 +1,82 @@
+import {EventEmitter} from 'node:events'
+import Redlock, {type Lock} from 'redlock'
+import {Logger} from './utils/Logger'
+import type RedisInstance from './utils/RedisInstance'
+
+type LeaderElectorEvents = {
+  elected: []
+  revoked: []
+  renewed: []
+}
+
+export class LeaderElector extends EventEmitter<LeaderElectorEvents> {
+  redis: RedisInstance
+  redlock: Redlock
+  lockKey: string
+  lockTTL: number
+  lock: Lock | null = null
+  renewTimer: NodeJS.Timeout | undefined = undefined
+  running: boolean = false
+  constructor(redis: RedisInstance, prefix: string, lockTTL: number) {
+    super()
+    this.redis = redis
+    this.redlock = new Redlock([this.redis], {retryCount: -1})
```

Comment:
-1 I'm not sure I totally understand what we're doing here. If we have a `retryCount: -1` a.k.a [infinity](https://github.com/mike-marcacci/node-redlock/blob/main/src/index.ts#L429-L430), then I would think that `tryAcquireLock()` would never return without a lock unless we run into some redis error.